File: src/main/java/ai/devchat/idea/setting/DevChatSettingsComponent.java
package ai.devchat.idea.setting;

import com.intellij.ui.components.JBLabel;
import com.intellij.ui.components.JBTextField;
import com.intellij.util.ui.FormBuilder;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

/**
 * Supports creating and managing a {@link JPanel} for the Settings Dialog.
 */
public class DevChatSettingsComponent {

    private final JPanel jPanel;
    private final JBTextField apiBaseText = new JBTextField();
    private final JBTextField apiKeyText = new JBTextField();
    private final JBTextField defaultModelText = new JBTextField();

    public DevChatSettingsComponent() {
        jPanel = FormBuilder.createFormBuilder()
                .addLabeledComponent(new JBLabel("api_base"), apiBaseText, 1, false)
                .addLabeledComponent(new JBLabel("api_key"), apiKeyText, 2, false)
                .addLabeledComponent(new JBLabel("default_model"), defaultModelText, 3, false)
                .addComponentFillVertically(new JPanel(), 0)
                .getPanel();
    }

    public JPanel getPanel() {
        return jPanel;
    }

    public JComponent getPreferredFocusedComponent() {
        return apiBaseText;
    }

    @NotNull
    public String getApiBase() {
        return apiBaseText.getText();
    }

    public String getApiKey() {
        return apiKeyText.getText();
    }

    public String getDefaultModel() {
        return defaultModelText.getText();
    }

    public void setApiBase(@NotNull String apiBase) {
        apiBaseText.setText(apiBase);
    }

    public void setApiKey(@NotNull String apiKey) {
        apiKeyText.setText(apiKey);
    }

    public void setDefaultModel(@NotNull String defaultModel) {
        defaultModelText.setText(defaultModel);
    }
}

File: src/main/kotlin/ai/devchat/idea/setting/DevChatSettingsComponent.kt
package ai.devchat.idea.setting

import com.intellij.openapi.ui.ComboBox
import com.intellij.ui.components.JBLabel
import com.intellij.ui.components.JBTextField
import com.intellij.util.ui.FormBuilder
import javax.swing.JComponent
import javax.swing.JPanel

/**
 * Supports creating and managing a [JPanel] for the Settings Dialog.
 */
class DevChatSettingsComponent {
    val panel: JPanel
    private val apiBaseText = JBTextField()
    private val apiKeyText = JBTextField()
    private val maxLogCountText = JBTextField()
    private val defaultModelText = ComboBox(arrayOf(
        "gpt-3.5-turbo",
        "gpt-3.5-turbo-1106",
        "gpt-3.5-turbo-16k",
        "gpt-4",
        "gpt-4-1106-preview",
        "claude-2",
        "xinghuo-2",
        "chatglm_pro",
        "ERNIE-Bot",
        "CodeLlama-34b-Instruct",
        "llama-2-70b-chat"
    ))

    init {
        panel = FormBuilder.createFormBuilder()
            .addLabeledComponent(JBLabel("api_base"), apiBaseText, 1, false)
            .addLabeledComponent(JBLabel("api_key"), apiKeyText, 2, false)
            .addLabeledComponent(JBLabel("max_log_count"), maxLogCountText, 3, false)
            .addLabeledComponent(JBLabel("default_model"), defaultModelText, 4, false)
            .addComponentFillVertically(JPanel(), 0)
            .panel
        defaultModelText.selectedItem = "gpt-3.5-turbo"
    }

    val preferredFocusedComponent: JComponent
        get() = apiBaseText
    var apiBase: String
        get() = apiBaseText.text
        set(apiBase) {
            apiBaseText.setText(apiBase)
        }
    var apiKey: String
        get() = apiKeyText.text
        set(apiKey) {
            apiKeyText.setText(apiKey)
        }
    var defaultModel: String
        get() = defaultModelText.selectedItem?.toString() ?: "gpt-3.5-turbo"
        set(defaultModel) {
            defaultModelText.selectedItem = defaultModel
        }

    var maxLogCount: Int
        get() = maxLogCountText.text.toIntOrNull() ?: 20
        set(value) {
            maxLogCountText.setText(value.toString())
        }
}
